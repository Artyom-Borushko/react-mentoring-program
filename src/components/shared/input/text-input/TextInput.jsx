import React from 'react';
import { useField } from 'formik';
import '../input.css';

export default function TextInput({ label, ...props }) {
  const [field, meta] = useField(props);

  return (
    <div className="input-wrapper">
      <label htmlFor={props.id || props.name}>{label}</label>
      <input className="text-input" {...field} {...props} />
      {meta.touched && meta.error ? (
        <div className="form-field-validation-error">{meta.error}</div>
      ) : null}
    </div>
  );
}
