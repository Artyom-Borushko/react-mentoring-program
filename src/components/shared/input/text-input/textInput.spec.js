import { render, screen } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { Formik } from 'formik';
import userEvent from '@testing-library/user-event';
import TextInput from './TextInput';

describe('Text input component tests', () => {
  it('renders text input with Formik', () => {
    const { container } = render(
      <Formik
        initialValues={{ movieTitle: '' }}
        onSubmit={() => {}}
      >
        <TextInput
          label="Title"
          name="movieTitle"
          type="text"
          id="movieTitle"
          placeholder="Title"
        />
      </Formik>,
    );

    expect(container).toMatchSnapshot();
  });

  it('can type in text input with Formik', async () => {
    render(
      <Formik
        initialValues={{ movieTitle: '' }}
        onSubmit={() => {}}
      >
        <TextInput
          label="Title"
          name="movieTitle"
          type="text"
          id="movieTitle"
          placeholder="Title"
        />
      </Formik>,
    );

    const titleInput = screen.getByLabelText('Title');
    await userEvent.click(titleInput);
    await userEvent.keyboard('test title 123');

    expect(titleInput.value).toBe('test title 123');
  });
});
