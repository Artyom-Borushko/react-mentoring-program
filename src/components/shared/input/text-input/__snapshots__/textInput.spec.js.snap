// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text input component tests renders text input with Formik 1`] = `
<div>
  <div
    class="input-wrapper"
  >
    <label
      for="movieTitle"
    >
      Title
    </label>
    <input
      class="text-input"
      id="movieTitle"
      name="movieTitle"
      placeholder="Title"
      type="text"
      value=""
    />
  </div>
</div>
`;
